src/features/users/components/users-overview.tsx

```diff
@@ -0,0 +1,206 @@
+    }
+    }, [dispatch, localUsersData])
+
+  const initialValuesAdd = {
+    email: '',
+    name: '',
+    status: '',
+    gender: 'male',
+  };
+
+  const checkIfIsValid = (value: any) =>
+    validationSchema.validate(value)
+    .then(() => setIsSaveButtonEnabled(true))
+    .catch(() => setIsSaveButtonEnabled(false));
+
+  const formik = useFormik({
+    initialValues: initialValuesAdd,
+    validationSchema,
+    validate: checkIfIsValid,
+    onSubmit: () => {},
+  });
+
+  const handleSaveNewUser = () => {
+    const postBody = {
+      name: formik.values?.name,
+      email: formik.values?.email,
+      gender: formik.values?.gender,
+      status: formik.values?.status,
+    } as User
+
+    dispatch(addUser(postBody))
+    setIsAdd(false);
+  }
+
+  const addNewUserView = (
+    <div>
+      <Box padding={2} paddingBottom={4}>
+        <Paper elevation={4}>
+          <Grid container spacing={2} paddingLeft={1} paddingRight={1}>
+            <Grid item xs={12}>
+              <Typography>
+                Add
+              </Typography>
+            </Grid>
+            <Grid item xs={6}>
+              <FormikInputField id={"email"} label={"Email"} formik={formik}/>
+            </Grid>
+            <Grid item xs={6}>
+              <FormikInputField id={"name"} label={"Name"} formik={formik}/>
+            </Grid>
+            <Grid item xs={12}>
+              <Divider />
+            </Grid>
+            <Grid item xs={6}>
+              <TextField
+                id="status"
+                name="status"
+                select
+                fullWidth
+                label="Status"
+                value={formik.values.status}
+                onChange={formik.handleChange}
+                onBlur={formik.handleBlur}
+                helperText={formik.touched.status && formik.errors.status}
+              >
+                {STATUSES.map((option: string) => (
+                  <MenuItem key={option} value={option}>
+                    {option}
+                  </MenuItem>
+                ))}
+              </TextField>
+            </Grid>
+            <Grid item xs={6}>
+              <TextField
+                id="gender"
+                name="gender"
+                select
+                fullWidth
+                label="Gender"
+                value={formik.values.gender}
+                onChange={formik.handleChange}
+                onBlur={formik.handleBlur}
+                helperText={formik.touched.gender && formik.errors.gender}
+              >
+                {GENDERS.map((option: string) => (
+                  <MenuItem key={option} value={option}>
+                    {option}
+                  </MenuItem>
+                ))}
+              </TextField>
+            </Grid>
+          </Grid>
+        </Paper>
+      </Box>
+    </div>
+  )
+
+
+  const addUserControl = (
+    <Grid container>
+      <Grid item xs={12}>
+        <IconButton onClick={() => {setIsAdd(false)}}>
```

Comment:
Best to avoid using the arrow function inside render. Because when you use arrow functions within render, each call to render will create new function objects. You can use either useCallback or .bind

Example:
`onClick={setIsAdd.bind(null, false)}`